### PR TITLE
major changes in Login and Registration pages

### PR DESCRIPTION
**Pull request by pbutiya**

### CODE REVIEW

File: front-end/src/views/Register.vue [hunk 1]
```diff
@@ -1,15 +1,108 @@
...
+    </div>
+    <div class="loginLink">
+    Already have an account? 
+    <router-link to = "/Login">Login here</router-link><br><br>
+    <router-link to = "/" class="home">Home</router-link>
   </div>
 </template>
 
-<style scoped>
+<script>
```
Comment (harjothkaur) [inline front-end/src/views/Register.vue:28]: looks good to me, but I request you to change the javascript style here in register
Here's the link where you can refer and change the style of javascript accordingly.
https://vuejs.org/guide/extras/composition-api-faq.html

**Review by harjothkaur — CHANGES_REQUESTED** [review]

File: front-end/src/views/Register.vue (hunk 1 above)
Comment (pbutiya) [reply front-end/src/views/Register.vue:28]: As we discussed i didnt write any code for specific registration page but i linked it with login page so in login page i already did javascript style.

File: front-end/src/components/Footer.vue [hunk 2]
```diff
@@ -22,12 +22,13 @@
 
 <style scoped>
 #footer {
-    background-color: #ffdfa8;
-    height: 100px;
```
Comment (austin-carnahan) [inline front-end/src/components/Footer.vue:26]: Could you explain the reason for removing the height property of the footer?

File: front-end/src/components/Footer.vue [hunk 3]
```diff
@@ -22,12 +22,13 @@
...
+    color: white;
+    padding: 10px;
+    font-size: small;
+    background-color: #767676;
     position: fixed;
-    left: 0;
     bottom: 0;
-    width: 100%;
+    width: 97.65%;
```
Comment (austin-carnahan) [inline front-end/src/components/Footer.vue:31]: I think this should be changed back to 100%. Why the change to 97.65%?

This is kind of giving me "magic number" vibes. What does this do? And can we think of a better way?

File: front-end/src/components/Navbar.vue [hunk 4]
```diff
@@ -19,11 +19,11 @@ function showMenu() {
             </div> 
             <div class="nav-content" :class="this.showMobileMenu ? 'open-menu' : 'closed-menu'"> 
             <div class="logo">
-                <router-link to="/">LOGO</router-link>
+                <router-link to="/">Home</router-link>
             </div> 
                 <ul class="nav-items">
                     <li>
-                        <router-link to="/multiple-choice">Multiple Choice Quiz</router-link>
```
Comment (austin-carnahan) [inline front-end/src/components/Navbar.vue:25]: I think there needs to be a link to the quiz either on the navbar or the homepage. How else will a user get to the quiz?

 Could you put this link back or add a new one on the homepage?

File: front-end/src/views/Login.vue [hunk 5]
```diff
@@ -1,15 +1,94 @@
...
+    <div class="regLink">
+    Don't have an account? 
+    <router-link to = "/Register">Register here<br></router-link>
+    <br>
+    <router-link to = "/" class="home">Home</router-link>
   </div>
 </template>
 
-<style scoped>
+<script>
```
Comment (austin-carnahan) [inline front-end/src/views/Login.vue:25]: See the conversation in `Register.vue`. Please change this javascript to Vue's Composition API style

File: front-end/src/views/Login.vue [hunk 6]
```diff
@@ -1,15 +1,94 @@
...
+  data() {
+    return {
+      username: '',
+      password: '',
+    }
+  },
+  methods: {
+    login() {
+      if (this.username && this.password) {
+      // Perform login logic here
```
Comment (austin-carnahan) [inline front-end/src/views/Login.vue:36]: Please add the login logic to this function. The API endpoints for this are up and running and there's good sample code for doing this in `MultipleChoiceQuiz.vue`.

Here is the code from that file, slightly modified. It should work here:

```
//POST request body
const options = {
    username: this.username,
    password: this.password
}
// send login data to the API
fetch("http://127.0.0.1:5000/login", {
  method: 'POST',
  body: JSON.stringify(options),
  headers: { 'Content-Type': 'application/json' }
}).then(res => res.json())
  .then(json => {
    console.log(json)
    window.location.href = '/multiple-choice'
  })
  .catch(err => console.log("Unable to login: " + err))
```

File: front-end/src/views/Login.vue [hunk 7]
```diff
@@ -1,15 +1,94 @@
...
+      // For example, send a request to a server
+      // with the username and password
+      window.location.href = '/multiple-choice'
```
Comment (austin-carnahan) [inline front-end/src/views/Login.vue:39]: Remove this redirect, place inside fetch logic with suggested code above

File: front-end/src/views/Register.vue [hunk 8]
```diff
@@ -1,15 +1,108 @@
...
+      password: '',
+      confirmPassword: '',
+    }
+  },
+  methods: {
+    register() {
+      if (this.username && this.email && this.password && this.confirmPassword && this.password == this.confirmPassword) 
+      {
+      // Perform registration logic here
+      // For example, send a request to a server
```
Comment (austin-carnahan) [inline front-end/src/views/Register.vue:43]: Add the logic to send the request to the server. As in `Login.vue` here's code I modified from `MultipleChoiceQuiz.vue`

```//POST request body
const options = {
    username: this.username,
    password: this.password
}
// Send register information to api
fetch("http://127.0.0.1:5000/register", {
  method: 'POST',
  body: JSON.stringify(options),
  headers: { 'Content-Type': 'application/json' }
}).then(res => res.json())
  .then(json => {
    console.log(json)
    window.location.href = '/multiple-choice'
  })
  .catch(err => console.log("Unable to register: " + err))
```

File: front-end/src/views/Register.vue [hunk 9]
```diff
@@ -1,15 +1,108 @@
...
+      // with the registration information
+      window.location.href = '/multiple-choice'
```
Comment (austin-carnahan) [inline front-end/src/views/Register.vue:45]: Remove this redirect, place inside fetch logic like example above

**Review by austin-carnahan — CHANGES_REQUESTED** [review]
Nice work! There's a few changes here that I don't really understand and had questions about. 

Other than that, I'd just like to see the Login/Registration components talk to the api server and for the javascript on those pages to be consistent with the style used elsewhere in the project (Vue's Composition API vs Options API). There should be resources and examples posted along with those comments. Let me know if you have questions

File: front-end/src/components/Footer.vue (hunk 2 above)
Comment (pbutiya) [reply front-end/src/components/Footer.vue:26]: If the height property was there then the page we can scroll up and down but there is no meaning of this so i fitted it with the desktop size thats why i removed footer.

File: front-end/src/components/Navbar.vue (hunk 4 above)
Comment (pbutiya) [reply front-end/src/components/Navbar.vue:25]: After logging in it will redirect to the quiz and i removed from the navbar because i think its not make any sense to put it on homepage because it its on homepage user can directly go to the quiz without logging in thats why i removed it.

File: front-end/src/views/Login.vue (hunk 5 above)
Comment (pbutiya) [reply front-end/src/views/Login.vue:25]: okay i will change it.

File: front-end/src/components/Footer.vue (hunk 2 above)
Comment (austin-carnahan) [reply front-end/src/components/Footer.vue:26]: Okay- makes sense to me!

File: front-end/src/views/Register.vue (hunk 1 above)
Comment (harjothkaur) [reply front-end/src/views/Register.vue:28]: please refer the above link so that you can make changes to the javascript style

File: front-end/src/components/Footer.vue (hunk 3 above)
Comment (austin-carnahan) [reply front-end/src/components/Footer.vue:31]: Resolving this per our conversation Tuesday. It fixes some issue with page elements and images mismatching. Can overhaul the css in a future PR.

File: front-end/src/views/Login.vue (hunk 6 above)
Comment (pbutiya) [reply front-end/src/views/Login.vue:36]: I created login page which will redirect and i created different page for login so here no need to write this code here as well.

File: front-end/src/views/Register.vue (hunk 1 above)
Comment (pbutiya) [reply front-end/src/views/Register.vue:28]: Okay i made changes and i will create a new PR.

File: front-end/src/views/Register.vue (hunk 9 above)
Comment (pbutiya) [reply front-end/src/views/Register.vue:45]: If its here there is no issue we can  redirect it with both way.

File: .vscode/launch.json [hunk 10]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (austin-carnahan) [inline .vscode/launch.json]: This file shouldnt be in the repo. You can add it to the .gitignore if it's something you need on your system

File: api/.Rhistory
Comment (austin-carnahan) [inline api/.Rhistory:1]: This also shouldn't be in the repo and can be gitignored if you need it

File: api/app/models/user.py [hunk 11]
```diff
@@ -1,6 +1,19 @@
+
 from app import db
 
 class User(db.Model):
     id = db.Column(db.Integer, primary_key=True, autoincrement=True)
     username = db.Column(db.String(50), unique=True)
-    password = db.Column(db.String(100))
\ No newline at end of file
+    password = db.Column(db.String(100))
```
Comment (austin-carnahan) [inline api/app/models/user.py:7]: These are new changes that need to be part of their own feature branch

**Review by austin-carnahan — APPROVED** [review]
going to go ahead and merge this. We can make any other changes needed later.
